controllers/tasks.js

```diff
@@ -0,0 +1,98 @@
+const Task = require('../models/Task')
+
+const addTask = (req, res) => {
+    res.render('pages/addTask')
+}
+
+const createTask = async (req, res) => {
+    try {
+        if (req.body.complete) {
+            req.body.completed=true
+        }
+        console.log(req.body, "Inside the create func")
+        await Task.create(req.body)
+        req.session.pendingMessage = "The task was created."
+        res.redirect('/tasks')
+    } catch (err) {
+        if (err.name === 'ValidationError') {
+            res.locals.message = Object.values(err.errors)
+                .map((item) => item.message)
+                .join(', ')
+        } else {
+            res.locals.message = "Something went wrong."
+        }
+        res.render('pages/addTask')
+    }
+}
+
+const deleteTask = async (req, res) => {
+    try {
+        const task = await Task.findByIdAndDelete(req.params.id)
+        req.session.pendingMessage = 'The task was deleted successfully.'
+        res.redirect('/tasks')
+    } catch (err) {
+        req.session.pendingMessage = 'Something went wrong.'
+        res.redirect('/tasks')
+    }
+}
+
+const editTask = async (req, res) => {
+    try {
+        const task = await Task.findById(req.params.id)
+        res.render('pages/editTask', { task })
+    } catch (err) {
+        req.session.pendingMessage = 'Something went wrong.'
+        res.redirect('/tasks')
+    }
+}
+
+const updateTask = async (req, res) => {
+    task = false
+    try {
+        if (req.body.complete) {         
```

Comment:
nit: I don't know if you can see this, but there's trailing whitespace at the end of this line. If you can't let me know; we can try to get your editor setup so that you can see trailing whitespaces & avoid having it there!

Reply:
My editor doesnt show any training white spaces but I see what your talking about.  Thats because I just used space bar. I use VS Code any idea how to get those visible. Generally I use Format document once am done coding so all this is taken care of.

Reply:
I'd try experimenting with this setting: https://stackoverflow.com/a/30140625/13262536